### PR TITLE
AdminPage

### PR DESCRIPTION
**Pull request by IWillSettYou**
Done with navigation

### CODE REVIEW

File: CourseController/src/router.js [hunk 1]
```diff
@@ -3,8 +3,25 @@ import { createMemoryHistory, createRouter } from 'vue-router'
 // TODO someday replace this with an auto generated router
 // pain in the butt to batch add new routes
 const routes = [
-    { path: '/', component: () => import("./views/Home.vue") },
-    { path: '/admin', component: () => import("./views/Admin.vue") },
+    { path: '/', component: () => import("./views/Home.vue")},
+    {
+        path: '/admin',
+        component: () => import("./views/Admin.vue"),
+        redirect: '/admin/modules',
```
Comment (TGyAkos) [inline CourseController/src/router.js:10]: So this will be the main view of the admin page then?

File: CourseController/src/views/Admin.vue [hunk 2]
```diff
@@ -27,21 +26,8 @@ const toggleAdmin = () => {
...
-              <AdminModule content="Szerkesztes" />
-              <AdminModule content="Bejegyzes" />
-              <AdminModule content="Feltoltes" />
-              <AdminModule content="Ora" />
-              <Button label="Switch admin" icon="pi pi-cog" @click="toggleAdmin"  />
-            </div>
-          </div>
-        </div>
-      </div>
+      <router-view/>
```
Comment (TGyAkos) [inline CourseController/src/views/Admin.vue:29]: Important, `<router-view>` and `<RouterView>` is not the same, there are some interoperability issues, or smth ask @AFekexd

File: CourseController/src/views/admin/AdminModules.vue [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<script>
```
Comment (TGyAkos) [inline CourseController/src/views/admin/AdminModules.vue:1]: This should be `<script setup>`

File: CourseController/src/views/admin/AdminModules.vue [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import AdminNavbar from "../../components/AdminNavbar.vue";
+import AdminModule from "../../components/AdminModule.vue";
+import {provide, ref} from "vue";
+
+export default {
```
Comment (TGyAkos) [inline CourseController/src/views/admin/AdminModules.vue:6]: So we don't have to do this.
This is the old way of doing it, it does the same thing, it's just harder to maintain it if we use both ways simultaneously

**Review by TGyAkos — CHANGES_REQUESTED** [review]
These are mostly good, but there are syntax issues, otherwise this is good quality code

File: CourseController/src/views/admin/AdminAssignments.vue
Comment (TGyAkos) [inline CourseController/src/views/admin/AdminAssignments.vue:1]: This is in the wrong place it should be under `components/`

File: CourseController/src/views/admin/AdminModules.vue
Comment (TGyAkos) [inline CourseController/src/views/admin/AdminModules.vue:1]: This is in the wrong place it should be under `components/`

File: CourseController/src/views/admin/AdminUsers.vue
Comment (TGyAkos) [inline CourseController/src/views/admin/AdminUsers.vue:1]: This is in the wrong place it should be under `components/`

**Review by TGyAkos — CHANGES_REQUESTED** [review]

**Comment by AFekexd** [comment]
NEXT TIME: merge to dev first, then testing and just after that PR to main :)

**Comment by TGyAkos** [comment]
> NEXT TIME: merge to dev first, then testing and just after that PR to main :)

DO NOT MERGE ANY PR-s WHEN A THERE ARE UNRESOLVED CHANGE REQUESTS

File: CourseController/src/views/Admin.vue (hunk 2 above)
Comment (IWillSettYou) [reply CourseController/src/views/Admin.vue:27]: donre

**Pull request merged**